src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java

```diff
@@ -688,9 +689,12 @@ public U getByUuidAndInitializeCollections(String uuid) {
             spec = spec.and(withNameOrEmailLike(pattern));
         }
 
-        final Long[] tenantIds = userSearch.getTenantIds();
-        if (tenantIds != null && tenantIds.length > 0) {
+        Set<Long> tenantIds = userSearch.getTenantIds();
+        if (!CollectionUtils.isEmpty(tenantIds)) {
             spec = spec.and(usersIn(tenantIds));
+            if (tenantIds.contains(-1L)) {
+                spec = spec.or(noTenants());
```

Comment:
suspicious or here

Reply:
reason is the query in the following form:  SELECT id FROM users_data u LEFT JOIN tenants_users tu ON tu.user_id = u.id WHERE tu.tenant_id IN (2) OR u.active_tenant_id IS NULL;